pkg/apis/apps/validation/validation_test.go

```diff
@@ -635,6 +651,23 @@ func TestValidateStatefulSet(t *testing.T) {
 				field.Invalid(field.NewPath("spec", "updateStrategy", "rollingUpdate", "maxUnavailable"), nil, ""),
 			},
 		},
+		{
+			name: "invalid ordinals.start " + enableStatefulSetStartOrdinal,
+			set: apps.StatefulSet{
+				ObjectMeta: metav1.ObjectMeta{Name: "abc-123", Namespace: metav1.NamespaceDefault},
+				Spec: apps.StatefulSetSpec{
+					PodManagementPolicy: apps.ParallelPodManagement,
+					UpdateStrategy:      apps.StatefulSetUpdateStrategy{Type: apps.RollingUpdateStatefulSetStrategyType},
+					Selector:            &metav1.LabelSelector{MatchLabels: validLabels},
+					Template:            validPodTemplate.Template,
+					Replicas:            3,
+					Ordinals:            &apps.StatefulSetOrdinals{Start: -2},
+				},
+			},
+			errs: field.ErrorList{
+				field.Invalid(field.NewPath("spec", "ordinals.start"), nil, ""),
+			},
+		},
```

Comment:
Test for ordinals.start exceeding replicas and error, see my other comment.

Reply:
There isn't an invariant that we need to enforce here, since start and replicas can scale independently. I think the API you're thinking of would have the following hypothetical documentation:

```
start enables a StatefulSet to skip replicas [0, .spec.ordinals.start) from being created. Out of .spec.replicas possible replicas, only replicas [.spec.ordinals.start, .spec.replicas) will be created. This effectively limits the number of replicas to .spec.replicas - .spec.ordinals.start. Note that an API invariant enforces .spec.ordinals.start <= .spec.relicas.
```

I didn't want to go with this approach, since it would change the meaning of .spec.replicas from the number of replicas, to the number of potential replicas.